fhttp/http.go

```diff
@@ -40,28 +40,32 @@ type Fetcher interface {
 	Fetch() (int, []byte, int)
 }
 
-// TODO: make this usable simultaneously by multiple requests (for instance not share global state like extra headers)
-
 var (
-	// ExtraHeaders to be added to each request.
-	extraHeaders http.Header
-	// Host is treated specially, remember that one separately.
-	hostOverride string
 	// BufferSizeKb size of the buffer (max data) for optimized client in kilobytes defaults to 32k.
 	BufferSizeKb = 128
 	// CheckConnectionClosedHeader indicates whether to check for server side connection closed headers.
 	CheckConnectionClosedHeader = false
-	// case doesn't matter for those 3
+	// 'constants', case doesn't matter for those 3
 	contentLengthHeader   = []byte("\r\ncontent-length:")
 	connectionCloseHeader = []byte("\r\nconnection: close")
 	chunkedHeader         = []byte("\r\nTransfer-Encoding: chunked")
 	// Start time of the server (used in debug handler for uptime).
 	startTime time.Time
 )
 
-func init() {
-	ResetHeaders()
-	extraHeaders.Add("User-Agent", userAgent)
+// NewHTTPOptions creates and initialize a HTTPOptions object.
+func NewHTTPOptions(url string) *HTTPOptions {
+	h := HTTPOptions{}
+	return h.Init(url)
+}
+
+// Init initializes the headers in an HTTPOptions (User-Agent).
+func (h *HTTPOptions) Init(url string) *HTTPOptions {
```

Comment:
does this need to be exposed?  can we just enforce calls through `NewHTTPOptions()`?

Reply:
it's to set / initialize options inside an existing variable/struct/memory